packages/template-retail-react-app/app/hooks/use-current-basket.js

```diff
@@ -24,13 +28,54 @@ export const useCurrentBasket = ({id = ''} = {}) => {
     const currentBasket =
         basketsData?.baskets?.find((basket) => basket?.basketId === id) || basketsData?.baskets?.[0]
 
+    // count the number of items in each shipment and rollup total
+    let totalItems = 0
+    const shipmentIdToTotalItems =
+        currentBasket?.productItems?.reduce((acc, item) => {
+            totalItems += item.quantity
+            acc[item.shipmentId] = acc[item.shipmentId] || 0
+            acc[item.shipmentId] += item.quantity
+            return acc
+        }, {}) ?? {}
+
+    // count the number on non-empty pickup and delivery shipments
+    // also count the number of delivery shipments missing an address or shipping method
+    let totalDeliveryShipments = 0
+    let totalPickupShipments = 0
+    const pickupStoreIds = []
+    let someShipmentsNeedAddress = false
+    let someShipmentsNeedShippingMethod = false
+    currentBasket?.shipments?.forEach((shipment) => {
+        if (shipmentIdToTotalItems[shipment.shipmentId]) {
+            if (storeLocatorEnabled && isPickupShipment(shipment)) {
+                totalPickupShipments += 1
+                pickupStoreIds.push(shipment.c_fromStoreId)
+            } else {
+                totalDeliveryShipments += 1
+                if (isAddressEmpty(shipment.shippingAddress)) {
+                    someShipmentsNeedAddress = true
```

Comment:
What does this variable name mean? 
It is a sentence here, and it does not give a clear indicator what it means compared the the value it receives.